lib/IAMClient.js

```diff
@@ -335,23 +335,42 @@ class VaultClient {
             Action: 'ListAccounts',
             Version: '2010-05-08',
         };
-        if (typeof options.marker !== 'undefined') {
-            const { marker } = options;
+        const {
+            marker,
+            maxItems,
+            filterKey,
+            filterKeyStartsWith,
+        } = options;
+        if (typeof marker !== 'undefined') {
             const conv = Number.isNaN(Number.parseInt(marker, 10));
             assert.notStrictEqual(conv || marker === '', true,
                 'Marker must be a number');
             assert(Number(marker) >= 0, 'Marker must be >= 0');
             data.Marker = marker;
         }
-        if (typeof options.maxItems !== 'undefined') {
-            assert.strictEqual(typeof options.maxItems, 'number',
+        if (typeof maxItems !== 'undefined') {
+            assert.strictEqual(typeof maxItems, 'number',
                 'maxItems need to be a number');
-            const conv = Number.isNaN(Number.parseInt(options.maxItems, 10));
+            const conv = Number.isNaN(Number.parseInt(maxItems, 10));
             assert.notStrictEqual(conv, true,
                 'maxItems must be a number');
-            assert(options.maxItems > 0 && options.maxItems <= 1000,
+            assert(maxItems > 0 && maxItems <= 1000,
                 'maxItems need to be a value between 1 and 1000 included');
-            data.MaxItems = options.maxItems;
+            data.MaxItems = maxItems;
+        }
+        if (filterKey && filterKeyStartsWith) {
+            assert(false,
+                'filterKey and filterKeyStartsWith parameters are exclusive');
+        }
+
+        if (filterKey && filterKey.length !== 0) {
+            assert(typeof filterKey === 'string', 'filter should be a string');
+            data.filterKey = filterKey;
+        }
+        if (filterKeyStartsWith && filterKeyStartsWith.length !== 0) {
+            assert(typeof filterKeyStartsWith === 'string',
+                'filter should be a string');
```

Comment:
Error message should be `filterKeyStartsWith` should be a string

Reply:
Changed in review commit: [`74a60d2` (#291)](https://github.com/scality/vaultclient/pull/291/commits/74a60d2633354ec5ce14c10e1d5644d0e3956d34)